<launch>
  <!-- CARALA rosbridge launch content starts-->
  <arg name='host' default=''/>
  <arg name='port' default=''/>
  <arg name='timeout' default=''/>
  <arg name='synchronous_mode' default=''/>
  <arg name='synchronous_mode_wait_for_vehicle_control_command' default=''/>
  <arg name='fixed_delta_seconds' default=''/>
  <arg name='town' default=''/>
  
  <rosparam file="$(find carla_ros_bridge)/config/settings.yaml" command="load" />
  <param name="carla/host" value="$(arg host)" unless="$(eval host == '')"/>
  <param name="carla/port" value="$(arg port)" unless="$(eval port == '')"/>
  <param name="carla/timeout" value="$(arg timeout)" unless="$(eval timeout == '')"/>
  <param
    name="carla/synchronous_mode"
    value="$(arg synchronous_mode)"
    unless="$(eval synchronous_mode == '')"/>
  <param
    name="carla/synchronous_mode_wait_for_vehicle_control_command"
    value="$(arg synchronous_mode_wait_for_vehicle_control_command)"
    unless="$(eval synchronous_mode_wait_for_vehicle_control_command == '')"/>
  <param
    name="carla/fixed_delta_seconds"
    value="$(arg fixed_delta_seconds)"
    unless="$(eval fixed_delta_seconds == '')"/>
  <param
    name="carla/town"
    value="$(arg town)"
    unless="$(eval town == '')"/>

  <node pkg="carla_ros_bridge" name="carla_ros_bridge" type="bridge.py" output="screen"/>
  <!-- CARALA rosbridge launch content ends-->
    
  <rosparam file="$(find parameter_optimisation)/config/general_settings.yaml" command="load" />
  <rosparam file="$(find parameter_optimisation)/config/scenario_settings.yaml" command="load" />

  <!-- Neural Architecture Settings-->
  <rosparam file="$(find parameter_optimisation)/config/nas_settings.yaml" command="load" />
  <arg name="sensor_definition_file" default="$(find parameter_optimisation)/config/sensors.json"/>
  <param name="sensor_definition_file" value='$(arg sensor_definition_file)'/>
  <node pkg="parameter_optimisation" type="parameter_optimisation.py" name="parameter_optimisation" output="screen">
  </node>
  
</launch>